feat: full models list in models.yaml

bo0tzz opened the pull request.

bo0tzz committed:
feat: full models list in models.yaml

bo0tzz committed:
feat: special case runner for large models (#16)

mertalev commented on models.yaml:
```diff
@@ -1,11 +1,137 @@
+    source: 'openclip'
+  - name: 'ViT-SO400M-14-SigLIP-384__webli'
+    source: 'openclip'
+    runner: 'mich'
```
Shouldn't all the SO400M models be run on mich? They're the same size

mertalev commented on models.yaml:
```diff
@@ -1,11 +1,137 @@
+    runner: 'mich'
+  - name: 'nllb-clip-base-siglip__v1'
+    source: 'openclip'
+  - name: 'nllb-clip-large-siglip__mrl'
```
This is the same size as the other nllb-large model

mertalev commented on models.yaml:
```diff
@@ -1,11 +1,137 @@
+    hf-name: 'xlm-roberta-base-ViT-B-32__laion5b_s13b_b90k'
     source: 'openclip'
-  - name: 'RN101__openai'
+  - name: 'XLM-Roberta-Large-ViT-H-14__frozen_laion5b_s13b_b90k'
```
I think this one is also 14GB

mertalev commented on models.yaml:
```diff
@@ -1,11 +1,137 @@
+    source: 'openclip'
+  - name: 'ViT-H-14-378-quickgelu__dfn5b'
+    source: 'openclip'
+  - name: 'ViT-H-14-quickgelu__dfn5b'
```
I'm not sure how big this one and its 378-sized cousin is, but IIRC they were slightly bigger than the SO400M models.

bo0tzz committed:
chore: put more models on mich (#17)

mertalev commented on immich_model_exporter/run.py:
```diff
@@ -103,11 +39,26 @@ def export_models(models: list[str], source: ModelSource) -> None:
+
+    export_models(models)
+
+    openclip_names = [m.hfName for m in models if m.source == ModelSource.OPENCLIP]
+    openclip_names_cleaned = [name.replace("__", ",") for name in openclip_names]
```
```suggestion
    openclip_names_cleaned = [m.hfName.replace("__", ",") for m in models if m.source == ModelSource.OPENCLIP]
```

bo0tzz replied:
Done

mertalev commented on immich_model_exporter/run.py:
```diff
@@ -1,92 +1,28 @@
-
+class ModelSpec(NamedTuple):
+  name: str
+  hfName: str
```
No camel case pls

bo0tzz replied:
I'm always so over the place I completely forget what conventions a particular language wants lol :sweat_smile:

bo0tzz committed:
fix: pr feedback, formatting

bo0tzz committed:
chore: use implicit HF_TOKEN env var

jrasm91 commented:
What is the status of this?

bo0tzz commented:
I was a bit scared to commit to having this system actually push into HF, and then I got distracted by other things. I've been meaning to pick it back up, but I'll probably at least wait until we have the new pokedex cluster going - as that will help minimize having this repo DoS our entire CICD runner infra.

mertalev commented:
I've been playing with some model export changes, so it's good timing to run it with those changes once we've got the cluster set up.

jrasm91 commented:
What's the status of this?

bo0tzz commented:
~~I moved them to the pokedex-large runner (8c/32G), there's only 2 concurrent ones of those, if needed we can change that (ideally by right-sizing a smaller runner that we can scale wider)~~

Moved from mich to pokedex runners, should now run much better as it won't all contend with eachother.
